tools/percolator/msgf2pin.xml

```diff
@@ -1,22 +1,56 @@
-<tool id="msgfplus2pin" name="MS-GF+ output to Pin converter" version="3.0">
+<tool id="msgfplus2pin" name="MS-GF+ output to Pin converter" version="3.0.1">
     <description>to create Percolator input files</description>
     <requirements>
         <requirement type="package" version="3.0">percolator</requirement>
     </requirements>
     <stdio>
         <exit_code range="1:"/>
     </stdio>
-    <command>
-             msgf2pin -o $pinout -e $enzyme
-             #if $matches
-             -m $matches
+    <command><![CDATA[
+             mkdir target && mkdir decoy &&
+             
+             #if $mzids.inputtype == "multi"
+                 #for $key in $mzids.target.keys()
+                     ln -s "$mzids.target[$key]" "target/$key" &&
+                     echo "target/$key" >> targetmeta &&
+                 #end for
+                 #for $key in $mzids.decoy.keys()
+                     ln -s "$mzids.decoy[$key]" "decoy/$key" &&
+                     echo "decoy/$key" >> decoymeta &&
+                 #end for
+                 msgf2pin -o $pinout -e $enzyme
+                 #if $matches
+                     -m $matches
+                 #end if
+                 $aafreq $ptm $pngase $isotope targetmeta decoymeta
+             
+             #else
+                 ln -s '$mzids.target' target/'$target.display_name' &&
+                 ln -s '$mzids.decoy' decoy/'$decoy.display_name' &&
```

Comment:
I would write `'decoy/$decoy.display_name'` not sure if your one is working, it looks strange at least :)